### PR TITLE
Update Application Insights Schema to support WorkItemConfig operations

### PR DESCRIPTION
**Pull request by debugthings**
Updating the componentWorkItemConfigs_API.json swagger def to address S360 work item. 

Changes
- Added **PATCH** and **GET** for a specific WorkItemConfig
- Added examples for both **PATCH** and **GET**
- Updated the **WorkItemConfigsCreate** (POST) example to match new schema
- Updated the **WorkItemCreateConfiguration** schema definition to match what is currently being sent by ARM and what is expected.
  - Changed **ValidateOnly** property to string from boolean to match current implementation
  - Changed **WorkItemProperties** to be a generic dictionary to match current implementation

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by debugthings** [comment]
@TomMilos can you look this over as well to make sure it's good for CDS?

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/6137 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/6137/commits/b87c384b6c15e4ecb18639125045e9545d4df640

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json [hunk 1]
```diff
@@ -173,15 +173,91 @@
...
             "$ref": "./examples/WorkItemConfigDelete.json"
           }
         }
+      },
+      "get": {
+        "description": "Gets specified work item configuration for an Application Insights component.",
+        "operationId": "WorkItemConfigurations_GetItem",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
```
Comment (dsgouda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json]: Are you trying to reference a definition from another RP here. That is not recommended since you will be taking a dependency on some other service's definition (even though resourcegroupparameter is fairly consistent across all RPs)
tl;dr just reference these parameters in this spec itself or create a common definition spec in this directory

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json [hunk 2]
```diff
@@ -173,15 +173,91 @@
...
+        "responses": {
+          "200": {
+            "description": "Successful operation indicating the config updated by this PATCH call.",
+            "schema": {
+              "$ref": "#/definitions/WorkItemConfiguration"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "WorkItemConfigurationsCreate": {
```
Comment (dsgouda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json]: Is this a long running operation

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json (hunk 2 above)
Comment (debugthings) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json]: It should be sub 20 seconds. But there is a chance it can run long.

Comment (dsgouda) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json]: In that case this should be marked as long running operation

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json (hunk 1 above)
Comment (debugthings) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json]: Agreed. I'll talk to our leads on this one as this specific pattern is all over this service definition. Maybe we could address it in a separate PR?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json (hunk 2 above)
Comment (debugthings) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json]: What's the bar for that? I'm going to look at my metrics and see what the breakdown of this operation is.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json (hunk 1 above)
Comment (dsgouda) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json]: @debugthings I take this back. Looks like `common-types` is a common definition schema for all RPs and this is the way to go forward. So this does look correct. Keep this unchanged. Apologize for the confusion 😄

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json (hunk 2 above)
Comment (dsgouda) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json]: I have marked this for ARM review, they will help you understand it better.

Comment (debugthings) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json]: Thanks. I'll add this note here for reference when the team reviews

99th percentile is about 5.8 seconds
90th percentile is 1 second
75th percentile is 630ms
50th percentile is 303ms

Comment (majastrz) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json]: Based on the data, I think it's reasonable to keep it as a synchronous operation. However, you said "it can run long". Can you clarify what that means?

---
In reply to: [281306078](https://github.com/Azure/azure-rest-api-specs/pull/5870#discussion_r281306078) [](ancestors = 281306078)

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json [hunk 3]
```diff
@@ -238,11 +314,14 @@
           "description": "Serialized JSON object for detailed properties"
         },
         "ValidateOnly": {
-          "type": "boolean",
-          "description": "Boolean indicating validate only"
+          "type": "string",
+          "description": "True or false string indicating validate only"
```
Comment (majastrz) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json]: Why are you changing a boolean property to a string one that takes "true" or "false"?

This would also be a breaking change to an API version marked as stable.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/WorkItemConfigGet.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  "parameters": {
+    "api-version": "2015-05-01",
+    "subscriptionId": "subid",
+    "resourceGroupName": "my-resource-group",
+    "resourceName": "my-component",
+    "workItemConfigId" : "Visual Studio Team Services"
+  },
+  "responses": {
+    "200": {
+      "body": {"ConnectorId":"d334e2a4-6733-488e-8645-a9fdc1694f41","IsDefault":true,"ConfigDisplayName":"Visual Studio Team Services","Id":"Visual Studio Team Services","ConfigProperties":"{\"VSOAccountBaseUrl\":\"https://testtodelete.visualstudio.com\",\"ProjectCollection\":\"DefaultCollection\",\"Project\":\"todeletefirst\",\"ResourceId\":\"d0662b05-439a-4a1b-840b-33a7f8b42ebf\",\"ConfigurationType\":\"Standard\",\"WorkItemType\":\"Bug\",\"AreaPath\":\"todeletefirst\",\"AssignedTo\":\"\"}"}
```
Comment (majastrz) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/WorkItemConfigGet.json]: >"body": {"ConnectorId":"d334e [](start = 6, length = 29)

This single line is a bit hard to read. Can you format it to match the rest of the JSON. (Same comment on the other examples.)

**Comment by majastrz** [comment]
>     "WorkItemCreateConfiguration": {

Just a heads-up, but this resource type schema is a violation of the RPC. Given that this was already available like this, I'm not going to block these changes.

However, ARM API review will block additions of new resource types that are not compliant with the RPC documented at [https://github.com/Azure/azure-resource-manager-rpc](https://github.com/Azure/azure-resource-manager-rpc)

---
Refers to: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json:304 in dba4561. [](commit_id = dba45619b38251efd81f7e3c35f4f9bd4cb77f6f, deletion_comment = False)

**Review by majastrz — CHANGES_REQUESTED** [review]
I added some comments. Please take a look.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json (hunk 2 above)
Comment (TomMilos) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json]: This is not a long running operation - it's just updating config in storage

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json [hunk 5]
```diff
@@ -173,15 +173,91 @@
...
+          "200": {
+            "description": "Successful operation indicating the config updated by this PATCH call.",
+            "schema": {
+              "$ref": "#/definitions/WorkItemConfiguration"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "WorkItemConfigurationsCreate": {
+            "$ref": "./examples/WorkItemConfigUpdate.json"
```
Comment (TomMilos) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json]: There is some inconsistency in naming of example files - in particular plural (...Config**s**...) for "Create" and "Get". Would be nice to make them consistent

```
./examples/WorkItemConfigDefaultGet.json	
./examples/WorkItemConfigDelete.json
./examples/WorkItemConfigsCreate.json
./examples/WorkItemConfigsGet.json
```

**Review by TomMilos — APPROVED** [review]

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json (hunk 5 above)
Comment (debugthings) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json]: There are now two net new operartions that get a specific (single) config. I agree the one S difference is confusing, but wanted to be consistent with the naming. Any suggestions to identify this as an operation on a single resource rather than multiple resources?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json (hunk 3 above)
Comment (debugthings) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/componentWorkItemConfigs_API.json]: I'm conflicted here since I wanted to make this represent what is being sent over the wire.

When capturing an example how the way Ibiza submits the data I noticed that we have been sending a string. When looking at our code we parse a string as a Boolean and return a Boolean as a string; however the model is using a Boolean. 

**Anyway, I'll revert this change since it's been this way for a very long time.**

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/WorkItemConfigGet.json (hunk 4 above)
Comment (debugthings) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/examples/WorkItemConfigGet.json]: Will do.

**Comment by debugthings** [comment]
@dsgouda and @majastrz I've completed the reviewed changes. Please check again and let me know if there are more fixes needed.

**Review by majastrz — APPROVED** [review]
Looks like all my comments were addressed. Thanks!

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
@debugthings Pleazse fix the model validation errors reported [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=28955)

**Comment by debugthings** [comment]
@dsgouda All good now it seems. Thanks for all your help
